### PR TITLE
Fix Mirror.cs implementation of SetTime to work on open files.

### PR DESCRIPTION
**Pull request by Deathspike**
This solves https://github.com/dokan-dev/dokan-dotnet/issues/136

### CODE REVIEW

File: sample/DokanNetMirror/Win32.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace DokanNetMirror
+{
+    public class Win32
```
Comment (magol) [inline sample/DokanNetMirror/Win32.cs]: - Please add XML comments to the class and method. See [NativeMethods.cs](https://github.com/dokan-dev/dokan-dotnet/blob/master/DokanNet/Native/NativeMethods.cs) for inspiration.
- Please rename the class to `NativeMethods`, see [CA1060: Move P/Invokes to NativeMethods class](https://msdn.microsoft.com/en-us/library/ms182161(v=vs.110).aspx)

File: sample/DokanNetMirror/Mirror.cs [hunk 2]
```diff
@@ -380,7 +380,18 @@ public NtStatus SetFileAttributes(string fileName, FileAttributes attributes, Do
         {
             try
             {
+                var stream = info.Context as FileStream;
+                if (stream != null)
+                {
+                    var ct = creationTime?.ToFileTime() ?? 0;
+                    var lat = lastAccessTime?.ToFileTime() ?? 0;
+                    var lwt = lastWriteTime?.ToFileTime() ?? 0;
+                    if (Win32.SetFileTime(stream.SafeFileHandle, ref ct, ref lat, ref lwt)) return NtStatus.Success;
```
Comment (magol) [inline sample/DokanNetMirror/Mirror.cs]: Please move the return to a new line.

You can have different tastes, but I would have preferred if it looked like this:
```C#
if (!Win32.SetFileTime(stream.SafeFileHandle, ref ct, ref lat, ref lwt))
{
    throw Marshal.GetExceptionForHR(Marshal.GetLastWin32Error());
}
return NtStatus.Success;

```

File: sample/DokanNetMirror/Mirror.cs [hunk 3]
```diff
@@ -151,7 +151,7 @@ private string GetPath(string fileName)
                         if (pathExists)
                         {
                             if (readWriteAttributes || pathIsDirectory)
-                            // check if driver only wants to read attributes, security info, or open directory
+                                // check if driver only wants to read attributes, security info, or open directory
```
Comment (magol) [inline sample/DokanNetMirror/Mirror.cs]: Maybe move the comment to above the `if`

File: sample/DokanNetMirror/Mirror.cs [hunk 4]
```diff
@@ -380,7 +380,18 @@ public NtStatus SetFileAttributes(string fileName, FileAttributes attributes, Do
         {
             try
             {
+                var stream = info.Context as FileStream;
+                if (stream != null)
```
Comment (magol) [inline sample/DokanNetMirror/Mirror.cs]:
```C#
var stream = info.Context as FileStream;
if (stream != null)
```
can be replaced width this in [C#7](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#pattern-matching)
```C#
if (info.Context is FileStream stream)
```

**Comment by Liryna** [comment]
@Deathspike thank you for the PR it is very welcome !
@magol has reviewed your PR with some pending questions. Can you look at them ?

**Comment by Liryna** [comment]
@magol do you think we can merge this an fix regarding your review ?

**Comment by magol** [comment]
@Liryna  Yes, I think we can merge this anyway :-)

**Comment by Liryna** [comment]
@magol Can you take care of it and apply changes of your reviews ❤️ ?

**Comment by Liryna** [comment]
Thank you @magol for the merge and fix !!! ❤️

**Pull request merged**